### PR TITLE
1263 api build debug

### PR DESCRIPTION
**Pull request by Jack-Hawthorne**
I've made some changes to the make.bat to generate fresh rst files every time a make clean or html is run This is to ensure new modules are accounted for and existing ones are updated.
I've also changed the docs config to include some special methods for which documentation was written but not created.

### CODE REVIEW

**Comment by daren-thomas** [comment]
Can one of the admins verify this patch?

**Comment by Jack-Hawthorne** [comment]
the specific build issues will be address one by one as i go through the documentation.

File: docs/conf.py [hunk 1]
```diff
@@ -195,3 +196,18 @@ def __getattr__(cls, name):
...
 
+
+# ## Add documentation for python special methods (with exclusions)
+
+def autodoc_skip_member(app, what, name, obj, skip, options):
+        exclusions = ('__weakref__',  # special-members
+                  '__doc__', '__module__', '__dict__',  # undoc-members
+                  )
+        exclude = name in exclusions
+        if name == name in exclusions:
```
Comment (daren-thomas) [inline docs/conf.py]: I don't understand this line.

File: docs/make.bat [hunk 2]
```diff
@@ -4,6 +4,9 @@ pushd %~dp0
 
 REM Command file for Sphinx documentation
 
+sphinx-apidoc -f -M -T -o modules ../cea
+print(Sphinx has updated rst files)
```
Comment (daren-thomas) [inline docs/make.bat]: did you check this? i'm not sure it can work.

Comment (daren-thomas) [reply docs/make.bat]: yep. batch doesn't have a `print(...)` statement. replace with:
```
echo Sphinx has updated rst files
```

Comment (Jack-Hawthorne) [reply docs/make.bat]: no worries ill have a look at these now

**Comment by Jack-Hawthorne** [comment]
@daren-thomas please have a look at the latest commit with your advised changes

File: cea/optimization/preprocessing/preprocessing_main.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 """
 ============================
-pre-processing algorithm
+Pre-processing algorithm
 ============================
```
Comment (JIMENOFONSECA) [inline cea/optimization/preprocessing/preprocessing_main.py]: @Jack-Hawthorne it is not clear why some of the lines  "===============" are erased and some others are not across different scripts. I think they should be erased from everywhere except where they are needed to be read by sphynxk. Would you mind to check for consistency?

**Comment by daren-thomas** [comment]
@Jack-Hawthorne you still have not addressed the question i had about `docs/conf.py` - come talk to me when you get to the office: I'd like to show you why I don't think this does what you think it does:

```
        if name == name in exclusions:
            return skip
```

**Comment by daren-thomas** [comment]
Also, this needs to get merged as soon as possible. I already had to do a lengthy merge to get it back on track with master.

**Pull request merged**